src/react-vite/05-routing/routing.md

```diff
@@ -12,130 +12,294 @@ user interface in response to URL changes.
+In the code above, it sets the `basename`, which is the base URL for all locations.
+Here, it's dynamically set based on the environment variable `import.meta.env.BASE_URL`.
+This is useful for scenarios where your application is served from a subdirectory on your server.
 
-```code
-/
-├── fruit
-    ├── apple
-    ├── banana
-```
+### RouterProvider
+
+Next, we can use `RouterProvider` to render our `<App>` component.
+Remember that our `main.tsx` file currently looks like this:
+
+@sourceref ../../../exercises/react-vite/05-routing/01-problem/src/main.tsx
+@highlight 11, only
+
+Instead of only rendering our `<App>` component, we can use `<RouterProvider>`:
+
+@sourceref ../../../exercises/react-vite/05-routing/01-solution/src/main.tsx
+@highlight 33, only
+
+The `RouterProvider` requires the `router` object created by `createBrowserRouter`.
+This object contains all the route configurations and settings for our application.
+
+### Outlet
+
+The final piece of React Router API that we need to use is the `<Outlet>`
+component.
 
-If the browser's URL path is `/fruit/apple/` and the component displaying the
-`fruit` portion of the path contains an `<Outlet>` then the `apple` component
-will be displayed. If the browser's URL path is `/fruit/orange/` then nothing
-will be displayed in the `<Outlet>` because there is no matching route.
+The `Outlet` component is used within a parent route component to render its
+child route components. Think of it as a marker that tells React Router,
+“Insert the child route component here.”
+
+When you define nested routes in your route configuration above, you don't
+immediately specify where in the parent component's JSX the child components
+should appear. Instead, you use `Outlet` in the parent component's JSX as a
+placeholder for where the matched child route component should be rendered.
+
+In the example below, the child component (`<AboutPage>` or `<HomePage>` from
+our `createBrowserRouter` example) would be rendered inside the `<main>` element:
+
+```tsx
+const Layout = () => {
+  return (
+    <>
+      <header>My awesome site</header>
+      <main>
+        <Outlet /> {/* Child routes will render here */}
+      </main>
+      <footer>©</footer>
+    </>
+  );
+};
+```
+@highlight 6
 
 ### Setup
 
-TODO
+To get starting with React Router in our application,
+let’s install `react-router-dom` with npm:
+
+✏️ Run:
 
 ```shell
 npm install react-router-dom@6
 ```
 
+✏️ Update **src/App.tsx** to be:
+
 @diff ../../../exercises/react-vite/04-components/02-solution/src/App.tsx ../../../exercises/react-vite/05-routing/01-problem/src/App.tsx only
 
+✏️ Update **src/main.tsx** to be:
+
 @diff ../../../exercises/react-vite/04-components/02-solution/src/main.tsx ../../../exercises/react-vite/05-routing/01-problem/src/main.tsx only
 
 ### Verify
 
+✏️ Update **src/App.test.tsx** to be:
+
 TODO
 
-### Exercise: Add routes and an Outlet
+### Exercise 1
```

Comment:
If I'm lost in the page I find having words for sections like Exercise more helpful than just a number.

Reply:
I changed this to be consistent with everything else and to shorten the fragment that shows up in the URL in our slide deck, but that said I *do* want to talk to you and learn more about how we can structure things to make it easier for everyone. I want to merge this in as-is but chat when we get a chance.